### DISCUSSION TITLE
Capacitor Android Issue (Custom Discord Authentication)

### DISCUSSION THREAD

**Discussion by gerardopar** [Web3Auth/Web3Auth#460] [Help]
We have a React/Capacitor application wired up to use web3Auth + Discord Login. We have used the following guide [Discord Guide](https://web3auth.io/docs/custom-authentication/social-providers/discord) to get setup & have it working successfully for **WEB** & **IOS**. But we are currently running into an issue with native android devices. 

At the moment when a user triggers the discord login from within our app the user is redirected to the following page (Open Login) & nothing happens, upon further investigation of the source maps we get an error about `redirectUrl` does not exist. But we have set that in our `adapterSettings`. 

https://user-images.githubusercontent.com/40479121/186069103-6254aee7-a275-4e95-829f-a66753e93c27.mov

<img width="1299" alt="Screen Shot 2022-08-22 at 9 01 53 PM" src="https://user-images.githubusercontent.com/40479121/186068090-49ed7c94-0f42-42ba-a121-59f63882b129.png">

Here is a snippet of our initialization code.

```javascript
const web3AuthInit = async () => {
    try {
      setInitLoading(true)
      const web3authCore = new Web3AuthCore({
        chainConfig: {
          chainNamespace: CHAIN_NAMESPACES.EIP155,
          chainId: '0x1',
          rpcTarget:
            'https://mainnet.infura.io/v3/776218ac4734478c90191dde8cae483c',
          blockExplorer: 'https://etherscan.io/',
          ticker: 'ETH',
          tickerName: 'Ethereum',
        },
        enableLogging: true,
      })

      const adapter = new OpenloginAdapter({
        adapterSettings: {
          network: 'testnet',
          clientId:
            'XXX',
          uxMode: 'redirect',
          redirectUrl: "XXX"
          loginConfig: {
            discord: {
              name: 'discord',
              verifier: 'discord-learn-card-verifier',
              typeOfLogin: 'discord',
              clientId: 'XXX',
              mainOption: true,
              showOnMobile: true,
              showOnDesktop: true,
            },
          },
        },
      })

      web3authCore.configureAdapter(adapter)
      subscribeAuthEvents(web3authCore)
      web3AuthStore.set.web3Auth(web3authCore)

      await web3authCore.init()
    } catch (error) {
      setInitLoading(false)
      console.log(error)
    }
  }
```

**Comment by shahbaz17** [comment]
Hey @gerardopar 

What's your `redirectUrl`? Also, have you whitelisted that in the dashboard?

And please note:

<img width="819" alt="Screenshot 2022-09-05 at 6 25 52 PM" src="https://user-images.githubusercontent.com/6962565/188454482-efbc6072-b388-470a-b5ae-14f37bdffbc3.png">
https://web3auth.io/docs/sdk/web/openlogin#openloginloginparams

**Comment by yashovardhan** [comment]
Please note that for Android, your URL looks similar to `{YOUR_APP_PACKAGE_NAME}://auth`

**Comment by TaylorBeeston** [comment]
@shahbaz17 @yashovardhan After doing an _immense_ amount of debugging, I've been able to localize the problem within the `@toruslabs/openlogin` package. It seems that, for some reason, in this specific environment, third party cookies _aren't_ working correctly!

To workaround this, I have set `no3PC` to `true`, but on the current version of `@web3auth/openlogin-adapter`, this will cause the adapter to fail to initialize due to an invalid check inside the `@toruslabs/openlogin` package. Fortunately, that has already been fixed (see [this commit](https://github.com/torusresearch/OpenLoginSdk/commit/5045f4bb0d2da4e33e8b6254ddeb9f81de216d2f)) upstream! The problem is that `@web3auth/openlogin-adapter` doesn't have the latest version of `@toruslabs/openlogin` bumped in its package json yet, so the code that ultimately gets pulled in still does not work.

I have put up [this PR](https://github.com/Web3Auth/Web3Auth/pull/564) to address this. Hopefully with that merged in and a new package released, this should start working! I have tested manually editing the compiled `@toruslabs/openlogin` file, and it has been working for me!

**Reply by yashovardhan** [reply]
That's really awesome. So glad you looked into this so deeply and submitted a PR as well. Further, @himanshuchawla009 and @chaitanyapotti can help you with reviewing the PR and understanding the concern.

Thanks a lot for taking the initiative to contribute 🔥

**Reply by TaylorBeeston** [reply]
@yashovardhan No problem! I'm just glad it turned out to be a pretty easy fix 😂

**Reply by Custard7** [reply]
Yes you rock @TaylorBeeston! 🔥 

@yashovardhan @himanshuchawla009 @chaitanyapotti thank you so much for helping to review this, will help us get our Android login flows off the ground 🚀 💪

**Reply by chaitanyapotti** [reply]
This is released in 2.1.2

**Comment by reslear** [comment]
hi, @gerardopar can you share an example of your implementation native+web?
